File: src/main/java/com/academy/model/Notification.java
package com.academy.model;

import lombok.Data;

import javax.persistence.*;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "notifications")
public class Notification {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long notificationId;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Type type;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String message;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Channel channel;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status;

    @Column
    private LocalDateTime sentAt;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    public enum Type {
        PAYMENT_REMINDER, ANNOUNCEMENT
    }

    public enum Channel {
        SMS, EMAIL, IN_APP
    }

    public enum Status {
        PENDING, SENT, FAILED
    }

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: src/main/resources/football-academy-frontend/src/app/models/notification.ts
import { User } from "./user";

export interface AppNotification {
  notificationId: number;
  user: User;
  type: 'PAYMENT_REMINDER' | 'ANNOUNCEMENT';
  message: string;
  channel: 'SMS' | 'EMAIL' | 'IN_APP';
  status: 'PENDING' | 'SENT' | 'FAILED';
  sentAt?: string;
  createdAt: string;
}
